dask/dataframe/shuffle.py

```diff
@@ -994,23 +994,44 @@ def fix_overlap(ddf, overlap):
     return new_dd_object(graph, name, ddf._meta, ddf.divisions)
 
 
-def compute_and_set_divisions(df, **kwargs):
-    mins = df.index.map_partitions(M.min, meta=df.index)
-    maxes = df.index.map_partitions(M.max, meta=df.index)
-    mins, maxes = compute(mins, maxes, **kwargs)
+def _compute_mins_and_maxes(column, **kwargs) -> tuple:
+    """For a given column, compute the min, max, and len of each partition.
+
+    And make sure that the partitions are sorted relative to each other.
+    NOTE: this does not guarantee that every partition is internally sorted.
+    """
+    mins = column.map_partitions(M.min, meta=column)
+    maxes = column.map_partitions(M.max, meta=column)
+    lens = column.map_partitions(len, meta=column)
+    mins, maxes, lens = compute(mins, maxes, lens, **kwargs)
     mins = remove_nans(mins)
     maxes = remove_nans(maxes)
-
+    non_empty_mins = [m for m, l in zip(mins, lens) if l != 0]
+    non_empty_maxes = [m for m, l in zip(maxes, lens) if l != 0]
```

Comment:
We don't care about empty partitions when computing current divisions because we will never be repartitioning. We guarantee that:
```
input.npartitions + 1 == len(divisions)

Reply:
So the idea here is that functions calling `_compute_partition_stats(allow_overlap=True)` are prepared for the fact that the number of mins/maxes might not match the number of divisions?

Reply:
yeah that's the idea. This seemed like the safest way to do the right thing when fixing overlap.